.github/workflows/cabal.project.local

```diff
@@ -1,6 +1,9 @@
-
 package cardano-crypto-praos
   flags: -external-libsodium-vrf
 
 program-options
   ghc-options: -Werror
+
+if impl(ghc >= 9.6)
+  program-options
+    ghc-options: -Wno-error=redundant-constraints
```

Comment:
I think we have module-local `-Wno-redundant-constraints` in `Ouroboros.Consensus.Util.EarlyExit` and `Test.Consensus.BlockchainTime.Simple`, can you remove them?

Also, can you add the respective Nix code to `nix/haskell.nix`:
```diff
diff --git a/nix/haskell.nix b/nix/haskell.nix
index 131e4cc7a..47eeaec7d 100644
--- a/nix/haskell.nix
+++ b/nix/haskell.nix
@@ -5,10 +5,10 @@ let
   inherit (final) haskell-nix;
 
   # from https://github.com/input-output-hk/haskell.nix/issues/298#issuecomment-767936405
-  forAllProjectPackages = cfg: { lib, ... }: {
+  forAllProjectPackages = cfg: args@{ lib, ... }: {
     options.packages = lib.mkOption {
       type = lib.types.attrsOf (lib.types.submodule ({ config, ... }: {
-        config = lib.mkIf config.package.isProject cfg;
+        config = lib.mkIf config.package.isProject (cfg args);
       }));
     };
   };
@@ -21,9 +21,13 @@ in
       "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
     };
     modules = [
-      (forAllProjectPackages {
-        ghcOptions = [ "-Werror" ];
-      })
+      (forAllProjectPackages ({ config, lib, ... }: {
+        ghcOptions = [ "-Werror" ] ++
+          lib.optional
+            # waiting for a new 9.6.x release to fix https://gitlab.haskell.org/ghc/ghc/-/issues/23323
+            (lib.hasPrefix "ghc96" config.compiler.nix-name)
+            "-Wno-error=redundant-constraints";
+      }))
       {
         # Options related to tasty-golden:
         packages.ouroboros-consensus-cardano.components.tests =
```

Reply:
Done